xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -0,0 +1,360 @@
+    initData.cryptoInfo.m_CryptoSessionId = hints.cryptoSession->sessionId;
+  }
+
+  initData.extraData = reinterpret_cast<const uint8_t*>(hints.extradata);
+  initData.extraDataSize = hints.extrasize;
+  initData.width = hints.width;
+  initData.height = hints.height;
+  initData.videoFormats = m_formats;
+
+  m_displayAspect = (hints.aspect > 0.0 && !hints.forced_aspect) ? hints.aspect : 0.0f;
+  m_width = hints.width;
+  m_height = hints.height;
+
+  m_processInfo.SetVideoDimensions(hints.width, hints.height);
+
+  return true;
+}
+
+bool CAddonVideoCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
+{
+  if (!m_struct.toAddon.Open)
+    return false;
+
+  unsigned int nformats(0);
+  for (auto fmt : options.m_formats)
+    if (fmt == RENDER_FMT_YUV420P)
+    {
+      m_formats[nformats++] = VideoFormatYV12;
+      break;
+    }
+  m_formats[nformats] = UnknownVideoFormat;
+
+  if (nformats == 0)
+    return false;
+
+  VIDEOCODEC_INITDATA initData;
+  if (!CopyToInitData(initData, hints))
+    return false;
+
+  m_lastPictureBuffer = nullptr;
+
+  return m_struct.toAddon.Open(m_addonInstance, initData);
+}
+
+bool CAddonVideoCodec::Reconfigure(CDVDStreamInfo &hints)
+{
+  if (!m_struct.toAddon.Reconfigure)
+    return false;
+
+  VIDEOCODEC_INITDATA initData;
+  if (!CopyToInitData(initData, hints))
+    return false;
+
+  return m_struct.toAddon.Reconfigure(m_addonInstance, initData);
+}
+
+bool CAddonVideoCodec::AddData(const DemuxPacket &packet)
+{
+  if (!m_struct.toAddon.AddData)
+    return false;
+
+  return m_struct.toAddon.AddData(m_addonInstance, packet);
+}
+
+CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPicture)
+{
+  if (!m_struct.toAddon.GetPicture)
+    return CDVDVideoCodec::VC_ERROR;
+
+  VIDEOCODEC_PICTURE picture;
+  picture.flags = (m_codecFlags & DVD_CODEC_CTRL_DRAIN) ? VIDEOCODEC_PICTURE::FLAG_DRAIN : 0;
+
+  switch (m_struct.toAddon.GetPicture(m_addonInstance, picture))
+  {
+  case VIDEOCODEC_RETVAL::VC_NONE:
+    return CDVDVideoCodec::VC_NONE;
+  case VIDEOCODEC_RETVAL::VC_ERROR:
+    return CDVDVideoCodec::VC_ERROR;
+  case VIDEOCODEC_RETVAL::VC_BUFFER:
+    return CDVDVideoCodec::VC_BUFFER;
+  case VIDEOCODEC_RETVAL::VC_PICTURE:
+    pVideoPicture->data[0] = picture.decodedData + picture.planeOffsets[0];
+    pVideoPicture->data[1] = picture.decodedData + picture.planeOffsets[1];
+    pVideoPicture->data[2] = picture.decodedData + picture.planeOffsets[2];
+    pVideoPicture->iLineSize[0] = picture.stride[0];
+    pVideoPicture->iLineSize[1] = picture.stride[1];
+    pVideoPicture->iLineSize[2] = picture.stride[2];
+    pVideoPicture->iWidth = picture.width;
+    pVideoPicture->iHeight = picture.height;
+    pVideoPicture->pts = DVD_NOPTS_VALUE;
+    pVideoPicture->dts = DVD_NOPTS_VALUE;
+    pVideoPicture->color_range = 0;
+    pVideoPicture->color_matrix = 4;
+    pVideoPicture->iFlags = DVP_FLAG_ALLOCATED;
```

Comment:
This part will be reworked at the time it is merged and rebased in vpupdates feature branch.
vpupdates branch contain a complete buffer rework, so it may this branch you are talking about?